src/wc/wc.rs

```diff
@@ -153,44 +212,79 @@ fn wc(files: Vec<String>, settings: &Settings) -> StdResult<(), i32> {
         let mut longest_line_length: usize = 0;
         let mut raw_line = Vec::new();
 
-        // reading from a TTY seems to raise a condition on, rather than return Some(0) like a file.
-        // hence the option wrapped in a result here
-        while match reader.read_until(LF, &mut raw_line) {
-            Ok(n) if n > 0 => true,
-            Err(ref e) if !raw_line.is_empty() => {
-                show_warning!("Error while reading {}: {}", path, e);
-                !raw_line.is_empty()
-            }
-            _ => false,
-        } {
-            // GNU 'wc' only counts lines that end in LF as lines
-            if *raw_line.last().unwrap() == LF {
-                line_count += 1;
+        #[cfg(unix)]
+        {
```

Comment:
Can you maybe split this out into a separate function?  It's kind of messy inlined like this.